consensus/state.go

```diff
@@ -1817,7 +1817,16 @@ func (cs *State) tryAddVote(vote *types.Vote, peerID p2p.ID) (bool, error) {
 					vote.Type)
 				return added, err
 			}
-			cs.evpool.AddEvidence(voteErr.DuplicateVoteEvidence)
+			var timestamp time.Time
+			if voteErr.VoteA.Height == 1 {
+				timestamp = cs.state.LastBlockTime // genesis time
+			} else {
+				timestamp = sm.MedianTime(cs.LastCommit.MakeCommit(), cs.LastValidators)
+			}
+			evidenceErr := cs.evpool.AddEvidence(types.NewDuplicateVoteEvidence(voteErr.VoteA, voteErr.VoteB, timestamp))
+			if evidenceErr != nil {
+				cs.Logger.Debug("Failed to add evidence to the evidence pool", "err", evidenceErr)
```

Comment:
This should probably be logged as an error, no?

Reply:
hmm yeah I think that would make sense